pkg/controller/controller.go

```diff
@@ -407,7 +434,7 @@ func (ctrl *resizeController) resizePVC(pvc *v1.PersistentVolumeClaim, pv *v1.Pe
 		ctrl.eventRecorder.Eventf(pvc, v1.EventTypeWarning, util.VolumeResizeFailed, err.Error())
 	}
 
-	return err
+	return ctrl.claims.Update(updatedPVC)
```

Comment:
We already update it in `resizeVolume`. Do we need to update it here too? Would it be simpler if we just always update it after immediately doing a successful api update?

Reply:
We did not update PVC in `resizeVolume` function. We updated PV in there.

Reply:
Same comment applies. Can we update the cache immediately after the update/patch operation, instead of the callers needing to realize that the function updated the api?

Reply:
I have moved code that updates the cache to be in same place where PV or PVC object is updated.